tests/unit/test_metrics.py

```diff
@@ -433,9 +436,84 @@ def test_sum_pairwise_distances_large_block_size(self) -> None:
         # Check output matches expected
         self.assertAlmostEqual(output, expected_output, places=5)
 
-    def test_mmd_block_ints(self) -> None:
+    def test_sum_pairwise_distances_zero_block_size(self):
+        """
+        Test sum_pairwise_distances when a zero block size is given.
+        """
+        # Setup some data
+        x = jnp.array([[0, 0], [1, 1], [2, 2]])
+        y = jnp.array([[0, 0], [1, 1]])
+
+        # Define a kernel object and set pairwise computations to be the square distance
+        kernel = MagicMock()
+        kernel.compute = coreax.util.squared_distance_pairwise
+
+        # Define a metric object
+        metric = coreax.metrics.MMD(kernel=kernel)
+
+        # Compute sum of pairwise distances with a zero block_size - the full text of
+        # the inbuilt error raised from the range function should provide all
+        # information needed for users to identify the issue
+        with self.assertRaises(ValueError) as error_raised:
+            metric.sum_pairwise_distances(x=x, y=y, block_size=0)
+        self.assertEqual(
+            error_raised.exception.args[0],
+            "range() arg 3 must not be zero",
+        )
+
+    def test_sum_pairwise_distances_negative_block_size(self):
+        """
+        Test sum_pairwise_distances when a negative block size is given.
+        """
+        # Setup some data
+        x = jnp.array([[0, 0], [1, 1], [2, 2]])
+        y = jnp.array([[0, 0], [1, 1]])
+
+        # Define a kernel object and set pairwise computations to be the square distance
+        kernel = MagicMock()
+        kernel.compute = coreax.util.squared_distance_pairwise
```

Comment:
We don't need to set `kernel.compute` if we're expecting to get an exception before it actually gets called

Reply:
(Same comment on the rest of the added tests)